src/transformers/modeling_utils.py

```diff
@@ -367,6 +367,10 @@ def shard_checkpoint(
         # check: https://github.com/huggingface/transformers/pull/24416 for more details
         if isinstance(weight, str):
             continue
+        elif weight.__class__.__name__ in ["QTensor"]:
```

Comment:
```suggestion
        elif weight.__class__.__name__ in ["QTensor", "QBitsTensor"]:
```

Reply:
Otherwise serializing 4-bit weights fails

Reply:
Thanks for the catch ! I will add tests for 4 bit